bugsnag-plugin-android-ndk/src/main/assets/include/event.h

```diff
@@ -163,6 +170,20 @@ bool bugsnag_event_is_unhandled(void *event_ptr);
 char *bugsnag_event_get_grouping_hash(void *event_ptr);
 void bugsnag_event_set_grouping_hash(void *event_ptr, char *value);
 
+/* Metadata */
+
+void bugsnag_event_add_metadata_double(void *event_ptr, char *section, char *name, double value);
+void bugsnag_event_add_metadata_string(void *event_ptr, char *section, char *name, char *value);
+void bugsnag_event_add_metadata_bool(void *event_ptr, char *section, char *name, bool value);
+
+void bugsnag_event_clear_metadata_section(void *event_ptr, char *section);
+void bugsnag_event_clear_metadata(void *event_ptr, char *section, char *name);
+
+bsg_metadata_t bugsnag_event_has_metadata(void *event_ptr, char *section, char *name);
```

Comment:
The use of this method means that we need to search metadata twice (once to check existence and once to return the value). Do you foresee that this will be a particularly useful public function? If not, I would suggest that we remove this and just use `bugsnag_get_metadata_value` (or rename to the simpler `bugsnag_get_metadata`).

Reply:
I think this has utility in that it allows users to check whether a key exists or not. `false` might be a valid return value for `bugsnag_event_get_metadata_bool`, or it might signify that the key doesn't exist. Without this method an end-user has no way of knowing.